feat: add a CONTRIBUTING.md (#14)
sonatype-nexus-community/community-project-template: CONTRIBUTING.md

# How to contribute

> It's great your here and reading this guide, because we need volunteers to help keep this project active and alive for the greater benefit of everyone!

Reviewer: ```suggestion
It's great you're here and reading this guide, because we need volunteers to help keep this project active and alive for the greater benefit of everyone!
```

- [Engaging with this project](#engaging-with-this-project)
- [Develpoment Guidelines](#develpoment-guidelines)
  - [Coding Conventions](#coding-conventions)
- [Testing](#testing)
- [Submitting Contributions](#submitting-contributions)

## Engaging with this project

Here are some important resources:
- [GitHub Issues](./issues) - a place for bugs to be raised and feature requests made

> - [GitHub Discussions](./discussions) - a place to discuss ideas or real-world usage

Reviewer: These are giving a 404. I'm not sure whether we can create relative links to these non-code pages.

Author: It was an idea - any thoughts on resolving this?

Reviewer: added a suggestion below... only thing I can think of is to add a TODO.

## Develpoment Guidelines

*Explain in detail what development standards must be adhered to when contributions are made.*

*You might include:*

> - Testing requirements

Reviewer: ```suggestion
*TODO: Explain in detail what development standards must be adhered to when contributions are made.*

*You might include:*
- Testing requirements
- Explanation of PR Checks
- What to expect from a PR reviewer/approver (such as expected time to response)
```

### Coding Conventions

> - Code style, convention & quality standards (and checks to run)

Reviewer: ```suggestion
- TODO: Explain code style, convention & quality standards (and any associated checks that will be run)
```

  
## Testing

> *Explain in detail:*

Reviewer: ```suggestion
*TODO: Explain in detail:*
```

# How to contribute

It's great you're here and reading this guide, because we need volunteers to help keep this project active and alive for the greater benefit of everyone!

- [Engaging with this project](#engaging-with-this-project)
- [Develpoment Guidelines](#develpoment-guidelines)
  - [Coding Conventions](#coding-conventions)
- [Testing](#testing)
- [Submitting Contributions](#submitting-contributions)

## Engaging with this project

Here are some important resources:
- [GitHub Issues](./issues) - a place for bugs to be raised and feature requests made

> - [GitHub Discussions](./discussions) - a place to discuss ideas or real-world usage

Reviewer: ```suggestion
TODO: Add links below to your repo's GitHub Issues and Discussions pages.

Here are some important resources:
- [GitHub Issues](#) - a place for bugs to be raised and feature requests made
- [GitHub Discussions](#) - a place to discuss ideas or real-world usage
```
